src/community/features-templating/features-templating-core/src/main/java/org/geoserver/featurestemplating/builders/SourceBuilder.java

```diff
@@ -18,9 +20,22 @@
 
     private Expression source;
 
-    public SourceBuilder(String key, NamespaceSupport namespaces) {
+    /**
+     * A SourceBuilder is managed when it not produce any output but simply call the evaluation of
+     * children builder.
+     */
+    protected boolean managed;
```

Comment:
That field was previously named as `features ` since before this case was bounded to the features array of json like output format. In theory a managed one is a template piece that is not writing anything because the templateWriter already did in the start template output and invoke only the children features evaluation.  I guess then that `hasNotOwnOutput` fit better